### PR TITLE
Populate README

### PR DESCRIPTION
**Pull request by jowong4**
Populated README with usage and dependency information

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,76 @@
 # Physlr: Construct a Physical Map from Linked Reads
+
+Physlr is a *de novo* physical map generater using linked reads (10X or stLFR). This physical
```
Comment (lcoombe) [inline README.md]:
```suggestion
Physlr generates *de novo* physical maps using linked reads (10X or stLFR). This physical
```

Comment (sjackman) [inline README.md]:
```suggestion
Physlr constructs a *de novo* physical map using linked reads from 10x Genomics or stLFR. This physical
```

File: README.md [hunk 2]
```diff
@@ -1 +1,76 @@
...
+map can then be used to scaffold an existing assembly to yield chromosomal level contiguity.
+
+## Build Status
+[![Build Status](https://dev.azure.com/bcgsc/btl/_apis/build/status/bcgsc.physlr?branchName=master)](https://dev.azure.com/bcgsc/btl/_build/latest?definitionId=1&branchName=master)
```
Comment (sjackman) [inline README.md]: I'm not personally a fan of build status badges in a README, just my 2c.

File: README.md [hunk 3]
```diff
@@ -1 +1,76 @@
...
+
+
+# Dependencies
+
+* [ntCard](https://github.com/bcgsc/ntCard)
+* [ntHits](https://github.com/bcgsc/ntHits)
+* gxx 5 or newer with openmp
```
Comment (sjackman) [inline README.md]:
```suggestion
* GCC 5 or newer with openmp
```

File: README.md [hunk 4]
```diff
@@ -1 +1,76 @@
...
+* Python 3.5 or newer and the following packages
+    * community
+    * networkx
+    * numpy
+    * scipy
+    * sklearn
+    * tqdm
+
+Additionally, we recommend using pypy3 over regular python3 for improved time performance.
```
Comment (sjackman) [inline README.md]:
```suggestion
Additionally, we recommend using pypy3 over regular python3 for speed.
```

File: README.md [hunk 5]
```diff
@@ -1 +1,76 @@
...
+# Running Physlr
+
+Generating Physlr Physical Map with stLFR reads
+===========================
+
+	cd ~/experiment
+	ln -s ~/physlr/data/Makefile
+    ls
+    Makefile    lr.fq.gz #stLFR   ref.fa
+    make physical-map lr=lr ref=ref minimizer_overlap=stLFR
```
Comment (sjackman) [inline README.md]: Use triple backticks to escape code blocks rather than indenting.

File: README.md [hunk 6]
```diff
@@ -1 +1,76 @@
...
+
+Scaffolding a draft assembly with Physlr Physical Map
+===========================
+
+	cd ~/experiment
+	ln -s ~/physlr/data/Makefile
+    ls
+    Makefile    lr.fq.gz #10X   ref.fa    draft.fa
+    make scaffolds lr=lr ref=ref draft=draft minimizer_overlap=10X
```
Comment (sjackman) [inline README.md]: Use triple backticks to escape code blocks rather than indenting.

File: README.md [hunk 7]
```diff
@@ -1 +1,76 @@
...
+
+* `lr.physlr.physical-map.path`: Paths of barcodes (backbones) describing chromosomal length sequences.
+* `lr.physlr.physical-map.ref.n10.paf.gz.*.pdf`: Various visualizations of backbones against the reference.
+* `draft.physlr.fa`: Physlr scaffolded assembly with physical-map.
+* `draft.physlr.quast.tsv`: Quast metrics using the Physlr scaffolded assembly as query against the reference.
+
+Testing Compiled Physlr executables
+===========================
+	cd ~/physlr/src
+    make check
```
Comment (sjackman) [inline README.md]: Use triple backticks to escape code blocks rather than indenting.

File: README.md [hunk 8]
```diff
@@ -1 +1,76 @@
...
+
+# Output files
+
+* `lr.physlr.physical-map.path`: Paths of barcodes (backbones) describing chromosomal length sequences.
+* `lr.physlr.physical-map.ref.n10.paf.gz.*.pdf`: Various visualizations of backbones against the reference.
+* `draft.physlr.fa`: Physlr scaffolded assembly with physical-map.
+* `draft.physlr.quast.tsv`: Quast metrics using the Physlr scaffolded assembly as query against the reference.
+
+Testing Compiled Physlr executables
+===========================
```
Comment (sjackman) [inline README.md]: Use `#` rather than `===`.

File: README.md [hunk 9]
```diff
@@ -1 +1,76 @@
...
+	cd ~/physlr/src
+    make check
+
+# Acknowledgements
+
+This projects uses:
+* [btl_bloomfilter](https://github.com/bcgsc/btl_bloomfilter) BTL C/C++ Common bloom filters for bioinformatics projects implemented by Justin Chu
+* [klib](https://github.com/attractivechaos/klib) A standalone and lightweight C library implemented by Attractive Chaos
```
Comment (sjackman) [inline README.md]: Is this link more appropriate? https://github.com/lh3/readfq/blob/master/readfq.py

File: README.md (hunk 2 above)
Comment (jowong4) [reply README.md]: I do not have a strong opinion either way but I just see a lot of repo with it so I added it to the README. Why are you not a fan of the badge?

File: README.md (hunk 9 above)
Comment (jowong4) [reply README.md]: My apologies, I missed that acknowledgement. The klib is required as we use it here for indexlr https://github.com/bcgsc/physlr/blob/master/src/vendor/kseq.h.

File: README.md [hunk 10]
```diff
@@ -1 +1,76 @@
...
+	ln -s ~/physlr/data/Makefile
+    ls
+    Makefile    lr.fq.gz #10X   ref.fa    draft.fa
+    make scaffolds lr=lr ref=ref draft=draft minimizer_overlap=10X
+
+# Output files
+
+* `lr.physlr.physical-map.path`: Paths of barcodes (backbones) describing chromosomal length sequences.
+* `lr.physlr.physical-map.ref.n10.paf.gz.*.pdf`: Various visualizations of backbones against the reference.
+* `draft.physlr.fa`: Physlr scaffolded assembly with physical-map.
```
Comment (lcoombe) [inline README.md]:
```suggestion
* `draft.physlr.fa`: Physlr scaffolded assembly using the physical map.
```

File: README.md [hunk 11]
```diff
@@ -1 +1,76 @@
...
+* `draft.physlr.quast.tsv`: Quast metrics using the Physlr scaffolded assembly as query against the reference.
```
Comment (lcoombe) [inline README.md]:
```suggestion
* `draft.physlr.quast.tsv`: Quast metrics comparing the Physlr scaffolded assembly against the reference.
```

File: README.md [hunk 12]
```diff
@@ -1 +1,76 @@
...
+
+# Running Physlr
+
+Generating Physlr Physical Map with stLFR reads
+===========================
+
+	cd ~/experiment
+	ln -s ~/physlr/data/Makefile
+    ls
+    Makefile    lr.fq.gz #stLFR   ref.fa
```
[the remark this answers is not in the document]
Comment (jowong4) [reply README.md]: I feel that is a bit excessive. Maybe just rename the files to `linkedreads.fq.gz` and `reference.fa` and so on

[the remark this answers is not in the document]
Comment (lcoombe) [reply README.md]: Hmm I disagree for a README -- I think better to spell things out really clearly and be more verbose. In my opinion it's as important or more important than the Makefile help page. 
(See Tigmint, ARCS github READMEs)

File: README.md [hunk 13]
```diff
@@ -1 +1,76 @@
...
+## Optional dependencies
+
+- [pigz](https://zlib.net/pigz/) for parallel gzip
+- [zsh](https://sourceforge.net/projects/zsh/) for reporting time and memory usage
+
+
+Compiling Physlr from source
+===========================
+
+	cd ~/physlr/src
```
[the remark this answers is not in the document]
Comment (jowong4) [reply README.md]: The intention behind the tildes, was just an example location. I suppose I could have done the `/path_to_install/` that other examples use

File: README.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (jowong4) [reply README.md]: I updated the usage example with a verbose explanation.

File: README.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (lcoombe) [reply README.md]: Hmm yeah I guess I thought that could be confusing for less experienced users, but that's just my two cents.

File: README.md [hunk 14]
```diff
@@ -1 +1,87 @@
...
+```
+cd src
+make
+```
+
+# Running Physlr
+
+## Generating Physlr Physical Map with stLFR reads
+
+To construct a physical map, you would need linked reads from 10X Genomics or stLFR. To visualize the correctness and contiguity of the physical map, you wouls also need a reference genome. In this example, the linked reads and reference genome are called `linkedReads.fq.gz` and `reference.fa`, respectively. The linked reads are from stLFR so we specify `minimizer_overlap=stLFR` to use the default value for stLFR reads.
```
Comment (lcoombe) [inline README.md]:
```suggestion
To construct a physical map, you need linked reads from 10X Genomics or stLFR. In addition, to visualize the correctness and contiguity of the physical map, you need a reference genome. 
In this example, the linked reads and reference genome are named `linkedReads.fq.gz` and `reference.fa`, respectively. The linked reads are from stLFR so we specify `minimizer_overlap=stLFR` to use the default parameter setting for stLFR reads.
```

File: README.md [hunk 15]
```diff
@@ -1 +1,87 @@
...
+
+```
+cd experiment
+ln -s data/Makefile
+ls
+Makefile    linkedReads.fq.gz #stLFR   reference.fa
+make physical-map lr=linkedReads ref=ref minimizer_overlap=stLFR
+```
```
Comment (lcoombe) [inline README.md]: This soft-link will be unnecessary once we change the name of the Makefile. Will we make that adjustment in a future PR?

Comment (jowong4) [reply README.md]: Yeah the adjustment will be with the physlr-make PR

File: README.md [hunk 16]
```diff
@@ -1 +1,87 @@
...
+
+## Scaffolding a draft assembly with Physlr Physical Map
+
+To scaffold a draft assembly, you would need linked reads from 10X Genomics or stLFR, and an existing assembly. To obtain Quast metrics for the Physlr scaffolded assembly, you wouls also need a reference genome. In this example, the linked reads, draft assembly, and reference genome are called `linkedReads.fq.gz`, `draft.fa`, `reference.fa`, respectively. The linked reads are from 10X so we specify `minimizer_overlap=10X` to use the default value for 10X reads.
```
Comment (lcoombe) [inline README.md]:
```suggestion
To scaffold a draft assembly, you need linked reads from 10X Genomics or stLFR, and an existing assembly. In addition, to calculate Quast summary metrics for the Physlr scaffolded assembly, you also need a reference genome. 
In this example, the linked reads, draft assembly, and reference genome are called `linkedReads.fq.gz`, `draft.fa`, `reference.fa`, respectively. The linked reads are from 10X Genomics so we specify `minimizer_overlap=10X` to use the default value for 10X reads.
```

File: README.md [hunk 17]
```diff
@@ -1 +1,87 @@
...
+
+```
+cd experiment
+ln -s data/Makefile
+ls
+Makefile    linkedReads.fq.gz #10X   reference.fa    draft.fa
+make scaffolds lr=linkedReads ref=reference draft=draft minimizer_overlap=10X
+```
+
+See the help page for further information.
```
Comment (lcoombe) [inline README.md]:
```suggestion
See the help page for further information.
```bin/physlr-make help```
```
Or similar.

File: README.md (hunk 15 above)
Comment (jowong4) [reply README.md]: may as well put it in this one too

File: README.md [hunk 18]
```diff
@@ -1 +1,87 @@
...
+
+# Output files
+
+* `lr.physlr.physical-map.path`: Paths of barcodes (backbones) describing chromosomal length sequences.
```
Comment (lcoombe) [inline README.md]:
```suggestion
* `lr.physlr.physical-map.path`: Paths of barcodes (backbones).
```
I'm hesitant to be so presumptuous when just describing the output files...

File: README.md (hunk 9 above)
Comment (sjackman) [reply README.md]: You can instead point to `kseq.h` in the `readfq` repository. Both libraries are by the same author. https://github.com/lh3/readfq/blob/master/kseq.h

File: README.md (hunk 2 above)
Comment (sjackman) [reply README.md]: A build status badge is mean to be developer-facing not user-facing, and `README.md` is a user-facing document.

File: README.md [hunk 19]
```diff
@@ -1 +1,88 @@
...
+- [pigz](https://zlib.net/pigz/) for parallel gzip
+- [zsh](https://sourceforge.net/projects/zsh/) for reporting time and memory usage
+
+
+# Compiling Physlr from source
+
+```
+cd src
+make
+```
```
Comment (sjackman) [inline README.md]: Let's add instructions to install Physlr. It should ideally be a single command. Makes sense to use `pip3` for this purpose I think. Autotools would be the alternative.
```sh
pip3 install --user git+https://github.com/bcgsc/physlr
```

File: README.md [hunk 20]
```diff
@@ -1 +1,88 @@
...
+* `lr.physlr.physical-map.ref.n10.paf.gz.*.pdf`: Various graphs showing the contiguity and correctness of the backbones with respect to the reference.
+* `draft.physlr.fa`: Physlr scaffolded assembly using the physical map.
+* `draft.physlr.quast.tsv`: Quast metrics comparing the Physlr scaffolded assembly against the reference.
+
+# Testing Compiled Physlr executables
+
+```
+cd src
+make check
+```
```
Comment (sjackman) [inline README.md]: You can delete this section `Testing Compiled Physlr executables`.

File: README.md (hunk 9 above)
Comment (jowong4) [reply README.md]: Oh, Attractive Chaos is Heng Li. Now, I know.

File: README.md (hunk 2 above)
Comment (jowong4) [reply README.md]: Hmm, I see. I will remove it then, but wouldn't the badge also give the user a source of confidence if he were to clone from master?

**Comment by jowong4** [comment]
@sjackman Hello Shaun, can you approve the PR for now? If you have other comments, I will address them down the line. We are trying to make an internal release for GSC use. Thanks.

File: README.md [hunk 21]
```diff
@@ -1 +1,77 @@
 # Physlr: Construct a Physical Map from Linked Reads
+
+Physlr constructs a *de novo* physical map using linked reads from 10X Genomics or stLFR. This physical map can then be used to scaffold an existing assembly to yield chromosomal level contiguity.
+
+# Dependencies
+
+* [ntCard](https://github.com/bcgsc/ntCard)
+* [ntHits](https://github.com/bcgsc/ntHits)
+* GCC 5 or newer with openmp
```
Comment (sjackman) [inline README.md]:
```suggestion
* GCC 5 or newer with OpenMP
```

File: README.md [hunk 22]
```diff
@@ -1 +1,77 @@
...
+- [pigz](https://zlib.net/pigz/) for parallel gzip
+- [zsh](https://sourceforge.net/projects/zsh/) for reporting time and memory usage
+
+
+# Compiling Physlr from source
+
+```
+pip3 install --user git+https://github.com/bcgsc/physlr
+cd src
+make
```
Comment (sjackman) [inline README.md]: I believe it's possible to modify `setup.py` to run `cd src && make install`, though I've not done it myself. In the mean time, please add
```
git clone https://github.com/bcgsc/physlr
cd physlr/src && make install
```
Please add an `install` rule to the `Makefile`. It should install to the location `$(DESTDIR)$(PREFIX)`. The default value of `DESTDIR` is the empty string. The default value of `PREFIX` is `$(HOME)/.local` (or possibly `/usr/local` but I prefer `$(HOME)/.local`. Executables are installed into `$(DESTDIR)$(PREFIX)/bin`.
Within the context of `setup.py`, executables should be installed into the directory that `setup.py` tells you to install them into.

File: README.md [hunk 23]
```diff
@@ -1 +1,77 @@
...
+```
+
+# Running Physlr
+
+## Generating Physlr Physical Map with stLFR reads
+
+To construct a physical map, you need linked reads from 10X Genomics or stLFR. In addition, to visualize the correctness and contiguity of the physical map, you need a reference genome.
+In this example, the linked reads and reference genome are called `linkedReads.fq.gz` and `reference.fa`, respectively. The linked reads are from stLFR so we specify `minimizer_overlap=stLFR` to use the default value for stLFR reads.
```
Comment (sjackman) [inline README.md]: I'd suggest calling this argument `chemistry=chromium` or `chemistry=stlfr` (no capitals).

File: README.md [hunk 24]
```diff
@@ -1 +1,77 @@
...
+
+```
+cd experiment
+ls
+linkedReads.fq.gz #stLFR   reference.fa
```
Comment (sjackman) [inline README.md]: This line looks weird. I'd remove lines 44 and 45.

File: README.md [hunk 25]
```diff
@@ -1 +1,77 @@
...
+
+## Scaffolding a draft assembly with Physlr Physical Map
+
+To scaffold a draft assembly, you need linked reads from 10X Genomics or stLFR, and an existing assembly. In addition, to calculate Quast summary metrics for the Physlr scaffolded assembly, you need a reference genome.
+In this example, the linked reads, draft assembly, and reference genome are called `linkedReads.fq.gz`, `draft.fa`, `reference.fa`, respectively. The linked reads are from 10X Genomics so we specify `minimizer_overlap=10X` to use the default value for 10X Genomics reads.
+
+```
+cd experiment
+ls
+linkedReads.fq.gz #10X   reference.fa    draft.fa
```
Comment (sjackman) [inline README.md]: Ditto.

File: README.md (hunk 23 above)
Comment (sjackman) [inline README.md]: Rename `linkedReads.fq.gz` to `linkedreads.fq.gz`. Capital letters in filenames aren't typical and can cause confusion.

File: README.md [hunk 26]
```diff
@@ -1 +1,77 @@
...
+bin/physlr-make scaffolds lr=linkedReads ref=reference draft=draft minimizer_overlap=10X
+```
+
+See the help page for further information.
+```bin/physlr-make help```
```
Comment (sjackman) [inline README.md]: Why triple back ticks on one line? That's unusual. Usually it's triple back ticks for a block or single back ticks for inline.

Comment (jowong4) [reply README.md]: Didn't know there was a distinction. Will change to single back ticks

File: README.md (hunk 23 above)
Comment (jowong4) [reply README.md]: I will change this in another PR as this requires makefile modifications

File: README.md (hunk 22 above)
Comment (jowong4) [reply README.md]: see #118 for `install` rule. I will look into `setup.py` at another time.

**Pull request merged**